src/output.c

```diff
@@ -194,14 +196,21 @@ new_output_notify(struct wl_listener *listener, void *data)
 	struct server *server = wl_container_of(listener, server, new_output);
 	struct wlr_output *wlr_output = data;
 
+	/*
+	 * Name virtual outputs.
+	 */
+	if (wlr_output_is_headless(wlr_output) && server->headless.pending_output_name[0] != '\0') {
+		wlr_output_set_name(wlr_output, server->headless.pending_output_name);
+	}
```

Comment:
Although we do not currently support any other way to add virtual outputs I'd still suggest `server->headless.pending_output_name[0] = '\0'` here so that we don't run into issues when we add other ways to add virtual outputs in the future.